src/ol_infrastructure/components/aws/sftp.py

```diff
@@ -0,0 +1,247 @@
+        :type opts: ResourceOptions
+
+        :rtype: SFTPServer
+        """
+        super().__init__(
+            "ol:infrastructure:aws:SFTPServer", sftp_config.server_name, None, opts
+        )
+
+        generic_resource_opts = ResourceOptions(parent=self).merge(opts)
+
+        # Create S3 bucket for SFTP backend
+        self.bucket = s3.BucketV2(
+            f"{sftp_config.server_name}-sftp-bucket",
+            bucket=sftp_config.bucket_name,
+            tags=sftp_config.tags,
+            opts=generic_resource_opts,
+        )
+
+        # Enable versioning on the bucket
+        s3.BucketVersioningV2(
+            f"{sftp_config.server_name}-sftp-bucket-versioning",
+            bucket=self.bucket.id,
+            versioning_configuration=s3.BucketVersioningV2VersioningConfigurationArgs(
+                status="Enabled"
+            ),
+            opts=generic_resource_opts,
+        )
+
+        # Block public access to the bucket
+        s3.BucketPublicAccessBlock(
+            f"{sftp_config.server_name}-sftp-bucket-public-access-block",
+            bucket=self.bucket.id,
+            block_public_acls=True,
+            block_public_policy=True,
+            ignore_public_acls=True,
+            restrict_public_buckets=True,
+            opts=generic_resource_opts,
+        )
+
+        # Create IAM role for Transfer Family service
+        self.transfer_service_role = iam.Role(
+            f"{sftp_config.server_name}-sftp-service-role",
+            assume_role_policy=json.dumps(
+                {
+                    "Version": "2012-10-17",
+                    "Statement": [
+                        {
+                            "Effect": "Allow",
+                            "Principal": {"Service": "transfer.amazonaws.com"},
+                            "Action": "sts:AssumeRole",
+                        }
+                    ],
+                }
+            ),
+            tags=sftp_config.merged_tags(
+                {"Name": f"{sftp_config.server_name}-sftp-service-role"}
+            ),
+            opts=generic_resource_opts,
+        )
+
+        # Create IAM policy for S3 access
+        self.s3_access_policy = iam.Policy(
+            f"{sftp_config.server_name}-sftp-s3-access-policy",
+            policy=self.bucket.arn.apply(
+                lambda bucket_arn: json.dumps(
+                    {
+                        "Version": "2012-10-17",
+                        "Statement": [
+                            {
+                                "Effect": "Allow",
+                                "Action": [
+                                    "s3:GetObject",
+                                    "s3:GetObjectVersion",
+                                    "s3:PutObject",
+                                    "s3:DeleteObject",
+                                    "s3:DeleteObjectVersion",
+                                ],
+                                "Resource": f"{bucket_arn}/*",
```

Comment:
If this is the policy attached to a given user account, and if we allow multiple accounts to access a given bucket, then we should probably scope the resource down to the home directory of each user so that they can't enumerate and manipulate data from other accounts.